### PR TITLE
Homework2: Pavel Klepcha

### PR DESCRIPTION
**Pull request by klep4a**

### CODE REVIEW

**Comment by klep4a** [comment]
Homework2-2: Pavel Klepcha

File: .gitignore [hunk 1]
```diff
@@ -106,4 +106,5 @@ venv.bak/
 #additional
 .idea/
 .tox/
-.gitignore/
\ No newline at end of file
+.gitignore/
```
Comment (belovmd) [inline .gitignore:109]: good work

File: src/Homework2/2.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+"""
+
+if __name__ == '__main__':
+    n = int(input())
+    arr = list(map(int, input().split()))
+max = arr[0]
+for i in arr:
+    if i >= max:
+        max = i
+for i in sorted(arr, reverse=True):
```
Comment (belovmd) [inline src/Homework2/2.py]: as arr is list we can use arr.sort

File: src/Homework2/2.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+Output Format
+
+Print the runner-up score.
+"""
+
+if __name__ == '__main__':
+    n = int(input())
+    arr = list(map(int, input().split()))
+max = arr[0]
```
Comment (belovmd) [inline src/Homework2/2.py]: max is reserved word!

File: src/Homework2/2.py [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+
+Output the integer number indicating the total number of occurrences
+of the substring in the original string.
+
+"""
+
+
+def count_substring(string, sub_string):
+    x = 0
+    for i in range(0, len(string) - len(sub_string) + 1, 1):
```
Comment (belovmd) [inline src/Homework2/2.py]: count = 0
while sub_string in string: 
count +=1
string = string[string.find(sub_string):]

File: src/Homework2/3.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+len(list) - количество элементов в списке
+"""
+
+n = 'каждый охотник желает знать, где? сидит фазан.'.split()
+
+# n = input().split()
+
+max_word = ''
+for i in n:
+    i = i.strip('.,?')
```
Comment (belovmd) [inline src/Homework2/3.py]: :;[]{}() . ?

File: src/Homework2/fib-temp-var.py [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+Выведите n-ое число Фибоначчи, используя только временные
+переменные, циклические операторы и условные операторы. n - вводится
+"""
+
+n = int(input())
+for i in range(n - 1):
+    if i == 0:
+        fib1, fib2 = 0, 1  # Надеюсь правильно понял
+    else:                  # временные переменные
+        fib = fib1 + fib2
```
Comment (belovmd) [inline src/Homework2/fib-temp-var.py]: fib1, fib2 = fib2, fib1+fib2

**Review by belovmd — COMMENTED** [review]
Great work!
please address these minor issues:

File: .gitignore (hunk 1 above)
Comment (klep4a) [reply .gitignore:109]: thanks

File: src/Homework2/2.py (hunk 2 above)
Comment (klep4a) [reply src/Homework2/2.py]: applied

File: src/Homework2/2.py (hunk 3 above)
Comment (klep4a) [reply src/Homework2/2.py]: fixed

File: src/Homework2/2.py (hunk 4 above)
Comment (klep4a) [reply src/Homework2/2.py]: string = string[string.find(sub_string):] >>> does infinite loop, but
string = string[string.find(sub_string) + 1:] >>> works!
Solved.

File: src/Homework2/3.py (hunk 5 above)
Comment (klep4a) [reply src/Homework2/3.py]: fixed

File: src/Homework2/fib-temp-var.py (hunk 6 above)
Comment (klep4a) [reply src/Homework2/fib-temp-var.py]: clear

File: src/Homework2/3.py [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+3. Найти самое длинное слово в введенном предложении.
+Учтите что в предложении есть знаки препинания.
+Подсказки:
+my_string.split([chars]) возвращает список строк.
+len(list) - количество элементов в списке
+"""
+
+import string
+n = 'каждый: охотник; {желает} знать, где? сидит [фазан].'.split()
+# n = input().split()
```
Comment (belovmd) [inline src/Homework2/3.py:11]: redundant comment

**Review by belovmd — COMMENTED** [review]
9,5

**Pull request closed without merging**